### ISSUE TITLE
Build CuPy against ROCm? 

### ISSUE THREAD

**Issue by leofang** [conda-forge/cupy-feedstock#11]
This issue won't be resolvable in the foreseeable future (at least I don't think so -- lots of extra work and time investment are needed), but it's still worth being kept track.

CuPy recently adds experimental support for ROCm (targeting AMD GPUs): https://docs-cupy.chainer.org/en/latest/install_rocm.html. 

Now, there's a `roctools` from Numba: https://anaconda.org/numba/roctools, but based on the Numba issue tracker and the download times, I doubt it's been extensively tested. Moreover, it doesn't provide the full ROCm software stack, just some LLVM bindings. Users still need to install the ROCm environment themselves: http://numba.pydata.org/numba-doc/latest/roc/overview.html#installation.

For CuPy + ROCm to work in conda, I imagine we need the following things to work:
- a ROCm compiler stack supported by Anaconda (as NVCC is now)
- a ROCm counterpart for `cudatoolkit` (can `roctools` be reused here?) 
- some docker images that support ROCm
- some ROCm-enabled CI infrastructure for building, testing and shipping the recipe

Disclaimer: I know nothing about ROCm. 🙂

**Comment by isuruf** [comment]
ROCM compiler stack is on conda-forge, https://github.com/conda-forge/staged-recipes/issues/10123

**Comment by leofang** [comment]
Cool, thanks for the info @isuruf! Any estimation for when the ROCm support will be up and running?

**Comment by isuruf** [comment]
According to cupy docs, we need `hipblas hipsparse rocrand`. I'm waiting on https://github.com/RadeonOpenCompute/clang-ocl/issues/18 to go further.

**Comment by leofang** [comment]
> According to cupy docs, we need `hipblas hipsparse rocrand`. 

A quick update: `rocthrust` is also needed now.

**Comment by leofang** [comment]
Due to 1. lack of interest from AMD, 2. lack of bandwidth, 3. potential conflict of interests, I am closing this issue as "won't fix".
